gsoc-2024-backend/src/main/java/backend/BackendVerticle.java

```diff
@@ -3,18 +3,71 @@
 import io.vertx.core.AbstractVerticle;
 import io.vertx.core.Promise;
 import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.client.WebClient;
 
 public class BackendVerticle extends AbstractVerticle {
 
+  private static final String API_URL = "https://icanhazdadjoke.com/";
+
   public static void main(String[] args) {
-    Vertx vertx = Vertx.vertx();
-    vertx.deployVerticle(new BackendVerticle())
-      .onFailure(Throwable::printStackTrace)
-      .onSuccess(v -> System.out.println("Deployed backend"));
+    Vertx.vertx().deployVerticle(new BackendVerticle(), result -> {
+      if (result.succeeded()) {
+        System.out.println("Deployed backend");
+      } else {
+        result.cause().printStackTrace();
+      }
+    });
   }
 
   @Override
   public void start(Promise<Void> startPromise) {
-    startPromise.complete();
+    Router router = Router.router(vertx);
+    router.get("/joke").handler(this::fetchJoke);
+
+    // Creating an HTTP server
+    HttpServer server = vertx.createHttpServer();
+
+    // Setting request handler for the server
+    server.requestHandler(router).listen(8081, result -> {
+      if (result.succeeded()) {
+        System.out.println("Backend server started successfully on port 8081");
+        startPromise.complete();
+      } else {
+        startPromise.fail(result.cause());
+      }
+    });
+  }
+
+  private void fetchJoke(RoutingContext routingContext) {
+    WebClient webClient = WebClient.create(vertx);
```

Comment:
This will create a `WebClient` for each request hitting the server. Can you fix this?

Reply:
Yes I fixed it by creating a single instance of WebClient and reused it across multiple requests.
<img width="704" alt="Screenshot 2024-03-26 at 22 54 28" src="https://github.com/tsegismont/vertx-gsoc-2024-assessment/assets/152805932/9d014512-c6b1-492a-a383-e4158a512aae">